Baseten on inference providers (#3494)
huggingface/blog: baseten.md

---
title: "Baseten on Hugging Face Inference Providers 🔥"
thumbnail: /blog/assets/inference-providers/welcome-baseten.jpg
authors:
  - user: celinah
  - user: sbrandeis
  - user: Wauplin
---

![banner image](https://huggingface.co/blog/assets/inference-providers/welcome-baseten.jpg)

# Baseten on Hugging Face Inference Providers 🔥

We're thrilled to share that **Baseten** is now a supported Inference Provider on the Hugging Face Hub!

Baseten joins our growing ecosystem, enhancing the breadth and capabilities of serverless inference directly on the Hub's model pages. Inference Providers are also seamlessly integrated into our client SDKs (for both JS and Python), making it super easy to use a wide variety of models with your preferred providers.

> [Baseten](https://baseten.com) is an AI infrastructure platform that covers serverless AI, training and more. With a catalog of 10 models, Baseten makes it easy for developers to integrate a wide range of AI capabilities into their applications with minimal setup.

Author: it's not 10 on their platform but 10 in ours, to be corrected by them as it doesn't say on their platform

> [Baseten](https://baseten.com) is an AI infrastructure platform that covers serverless AI, training and more. With a catalog of 10 models, Baseten makes it easy for developers to integrate a wide range of AI capabilities into their applications with minimal setup.

Reviewer: ```suggestion
[Baseten](https://baseten.com) is an AI infrastructure platform that serves high-performance, scalable inference for open-source, custom, and fine-tuned models. With a catalog of many frontier models, Baseten makes it easy for developers to integrate a wide range of AI capabilities into their applications with minimal setup.
```

## How it works

### In the website UI

1. In your user account settings, you are able to:

- Set your own API keys for the providers you've signed up with. If no custom key is set, your requests will be routed through HF.
- Order providers by preference. This applies to the widget and code snippets in the model pages.

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/user-setting-v2.png" alt="Inference Providers"/>

2. As mentioned, there are two modes when calling Inference Providers:

- Custom key (calls go directly to the inference provider, using your own API key of the corresponding inference provider)
- Routed by HF (in that case, you don't need a token from the provider, and the charges are applied directly to your HF account rather than the provider's account)

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/explainer.png" alt="Inference Providers"/>

3. Model pages showcase third-party inference providers (the ones that are compatible with the current model, sorted by user preference)

> <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/model-widget-v2.png" alt="Inference Providers"/>

Reviewer 2: ```suggestion
<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/model-widget-v3.png" alt="Inference Providers"/>
```
Should be a nice upgrade!

### From the client SDKs

> baseten is available through the Hugging Face SDKs - `huggingface_hub` (>= 1.11.2) for Python and `@huggingface/inference` for JavaScript.

Reviewer: ```suggestion
Baseten is available through the Hugging Face SDKs - `huggingface_hub` (>= 1.11.2) for Python and `@huggingface/inference` for JavaScript.
```

> The following examples show how to use the latest [DeepSeek V4 Flash](deepseek-ai/DeepSeek-V4-Flash-0731) through baseten. Use a [Hugging Face token](https://huggingface.co/settings/tokens) to authenticate - the request will be routed to baseten automatically.

Reviewer: ```suggestion
The following examples show how to use the latest [DeepSeek V4 Flash](deepseek-ai/DeepSeek-V4-Flash-0731) through Baseten. Use a [Hugging Face token](https://huggingface.co/settings/tokens) to authenticate - the request will be routed to Baseten automatically.
```

- Order providers by preference. This applies to the widget and code snippets in the model pages.

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/inference-providers-baseten.png" alt="Inference Providers"/>

2. As mentioned, there are two modes when calling Inference Providers:

- Custom key (calls go directly to the inference provider, using your own API key of the corresponding inference provider)
- Routed by HF (in that case, you don't need a token from the provider, and the charges are applied directly to your HF account rather than the provider's account)

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/explainer.png" alt="Inference Providers"/>

3. Model pages showcase third-party inference providers (the ones that are compatible with the current model, sorted by user preference)

<img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/inference-providers/model-widget-v3.png" alt="Inference Providers"/>

### From the client SDKs

> Baseten is available through the Hugging Face SDKs - `huggingface_hub` (>= 1.11.2) for Python and `@huggingface/inference` for JavaScript.

Reviewer 3: ```suggestion
Baseten is available through the Hugging Face SDKs - `huggingface_hub` (>= 1.26.1) for Python and `@huggingface/inference` for JavaScript.
```
